src/side-nav/types.ts

```diff
@@ -0,0 +1,23 @@
+interface SideNavBaseProps {
+    "data-testid"?: string | undefined;
+    className?: string | undefined;
+    id?: string | undefined;
+    title: string;
+    children: React.ReactNode;
+}
+
+export interface SideNavProps
+    extends Pick<SideNavBaseProps, "children" | "className" | "data-testid"> {}
```

Comment:
with all the picks and omits, i suggest just being explicit about the prop attributes. Only merge them if u find commonalities.